compiler/luci-interpreter/src/kernels/Tanh.test.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ * Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "kernels/Tanh.h"
+#include "kernels/TestUtils.h"
+
+namespace luci_interpreter
+{
+namespace kernels
+{
+namespace
+{
+
+using namespace testing;
+
+TEST(TanhTest, Float)
+{
+  Shape input_shape{1, 2, 4, 1};
+  std::vector<float> input_data{
+      0, -6, 2,  4, //
+      3, -2, 10, 1, //
+  };
+  Tensor input_tensor = makeInputTensor<DataType::FLOAT32>(input_shape, input_data);
+  Tensor output_tensor = makeOutputTensor(DataType::FLOAT32);
+
+  Tanh kernel(&input_tensor, &output_tensor);
+  kernel.configure();
+  kernel.execute();
+
+  std::vector<float> ref_output_data{
+      0,          -0.9999877, 0.9640275, 0.999329,  //
+      0.99505475, -0.9640275, 1,         0.7615941, //
+  };
+  EXPECT_THAT(extractTensorData<float>(output_tensor),
+              ElementsAreArray(ArrayFloatNear(ref_output_data)));
+}
+
+TEST(TanhTest, Uint8)
+{
+  float scale = 1.f / 128;
+  int32_t zero_point = 0;
+  float kMin = -1;
+  // float kMin = 0;
+  float kMax = 127.f / 128.f;
+  // float kMax = 255;
+  float kTanhTolerance = 2 * (1. / 256);
+  std::pair<float, int32_t> quant_param = quantizationParams<uint8_t>(kMin, kMax);
+  std::vector<float> input_data{
+      0,  -6, 2, 4, //
+      -4, -2, 8, 1, //
+      0,  -6, 2, 4, //
+      -4, -2, 8, 1, //
+      0,  -6, 2, 4, //
+      -4, -2, 8, 1, //
+      0,  -6, 2, 4, //
+      -4, -2, 8, 1, //
+      0,  -6, 2, 4, //
+      -4, -2, 8, 1, //
+      0,  -6, 2, 4, //
+      -4, -2, 8, 1, //
+  };
+  Tensor input_tensor{
+      DataType::U8, {2, 6, 4, 1}, {{quant_param.first}, {quant_param.second}}, ""};
```

Comment:
The input tensor's quantization parameters do not match with the input data in line 80. Please try with
```
{{quant_param.first * 8}, {quant_param.second * 8}}
```

Reply:
<details><summary>Error Message</summary>
<pre><code>
Value of: dequantize<uint8_t>(extractTensorData<uint8_t>(output_tensor), output_tensor.scale(), output_tensor.zero_point())
Expected: has 48 elements where
element #0 is approximately 0 (absolute error <= 0.0078125),
element #1 is approximately -0.99998701 (absolute error <= 0.0078125),
element #2 is approximately 0.96402699 (absolute error <= 0.0078125),
element #3 is approximately 0.99932897 (absolute error <= 0.0078125),
element #4 is approximately -0.99932897 (absolute error <= 0.0078125),
element #5 is approximately -0.96402001 (absolute error <= 0.0078125),
element #6 is approximately 0.99998999 (absolute error <= 0.0078125),
element #7 is approximately 0.76159 (absolute error <= 0.0078125),
element #8 is approximately 0 (absolute error <= 0.0078125),
element #9 is approximately -0.99998701 (absolute error <= 0.0078125),
element #10 is approximately 0.96402699 (absolute error <= 0.0078125),
element #11 is approximately 0.99932897 (absolute error <= 0.0078125),
element #12 is approximately -0.99932897 (absolute error <= 0.0078125),
element #13 is approximately -0.96402001 (absolute error <= 0.0078125),
element #14 is approximately 0.99998999 (absolute error <= 0.0078125),
element #15 is approximately 0.76159 (absolute error <= 0.0078125),
element #16 is approximately 0 (absolute error <= 0.0078125),
element #17 is approximately -0.99998701 (absolute error <= 0.0078125),
element #18 is approximately 0.96402699 (absolute error <= 0.0078125),
element #19 is approximately 0.99932897 (absolute error <= 0.0078125),
element #20 is approximately -0.99932897 (absolute error <= 0.0078125),
element #21 is approximately -0.96402001 (absolute error <= 0.0078125),
element #22 is approximately 0.99998999 (absolute error <= 0.0078125),
element #23 is approximately 0.76159 (absolute error <= 0.0078125),
element #24 is approximately 0 (absolute error <= 0.0078125),
element #25 is approximately -0.99998701 (absolute error <= 0.0078125),
element #26 is approximately 0.96402699 (absolute error <= 0.0078125),
element #27 is approximately 0.99932897 (absolute error <= 0.0078125),
element #28 is approximately -0.99932897 (absolute error <= 0.0078125),
element #29 is approximately -0.96402001 (absolute error <= 0.0078125),
element #30 is approximately 0.99998999 (absolute error <= 0.0078125),
element #31 is approximately 0.76159 (absolute error <= 0.0078125),
element #32 is approximately 0 (absolute error <= 0.0078125),
element #33 is approximately -0.99998701 (absolute error <= 0.0078125),
element #34 is approximately 0.96402699 (absolute error <= 0.0078125),
element #35 is approximately 0.99932897 (absolute error <= 0.0078125),
element #36 is approximately -0.99932897 (absolute error <= 0.0078125),
element #37 is approximately -0.96402001 (absolute error <= 0.0078125),
element #38 is approximately 0.99998999 (absolute error <= 0.0078125),
element #39 is approximately 0.76159 (absolute error <= 0.0078125),
element #40 is approximately 0 (absolute error <= 0.0078125),
element #41 is approximately -0.99998701 (absolute error <= 0.0078125),
element #42 is approximately 0.96402699 (absolute error <= 0.0078125),
element #43 is approximately 0.99932897 (absolute error <= 0.0078125),
element #44 is approximately -0.99932897 (absolute error <= 0.0078125),
element #45 is approximately -0.96402001 (absolute error <= 0.0078125),
element #46 is approximately 0.99998999 (absolute error <= 0.0078125),
element #47 is approximately 0.76159 (absolute error <= 0.0078125)
  Actual: { -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, ... }, whose element #0 doesn't match, which is -1 from 0
</code></pre></details>

Thank you for your advice, @jinevening ,
I followed your advice

`Actual: { -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, ... }, `
Sadly, multipying 8 to quant_params doesn't match the output.

without multiplying 8 to quant_params, I get
`Actual: { 0, -0.757812, 0.757812, 0.757812, -0.757812, -0.757812, 0.757812, 0.757812, 0, -0.757812, 0.757812, 0.757812, -0.757812, -0.757812, 0.757812, 0.757812, 0, -0.757812, 0.757812, 0.757812, -0.757812, -0.757812, 0.757812, 0.757812, 0, -0.757812, 0.757812, 0.757812, -0.757812, -0.757812, 0.757812, 0.757812, ... }, whose element #1 doesn't match, which is 0.242175 from -0.999987`

I should work more with parameters.
Once again, thank you for your advice, @jinevening 
I really appreciate your concern on my works.
I will work harder and better for ONE project.